src/Core/Managed/Shared/DataContracts/AvailabilityTelemetry.cs

```diff
@@ -0,0 +1,170 @@
+﻿namespace Microsoft.ApplicationInsights.DataContracts
+{
+    using System;
+    using System.Collections.Generic;
+    using Microsoft.ApplicationInsights.Channel;
+    using Microsoft.ApplicationInsights.Extensibility.Implementation;
+    using Microsoft.ApplicationInsights.Extensibility.Implementation.External;
+    using System.Globalization;
+
+    /// <summary>
+    /// Telemetry type used for availability test results.
+    /// Contains a time and message and optionally some additional metadata.
+    /// </summary>
+    public sealed class AvailabilityTelemetry : ITelemetry, ISupportProperties //TODO: Should this implement IOperation like RequestTelemetry?
+    {
+        internal const string TelemetryName = "Availability";
+
+        internal readonly string BaseType = typeof(AvailabilityData).Name;
+        internal readonly AvailabilityData Data;
+        private readonly TelemetryContext context;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="AvailabilityTelemetry"/> class with empty properties.
+        /// </summary>
+        public AvailabilityTelemetry()
+        {
+            this.Data = new AvailabilityData();
+            this.context = new TelemetryContext(this.Data.properties);
+            this.Data.testRunId = Guid.NewGuid().ToString();
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="AvailabilityTelemetry"/> class with empty properties.
+        /// </summary>
+        public AvailabilityTelemetry(string testName, DateTimeOffset testTimeStamp, TimeSpan duration, string runLocation, bool success, string message = null)
+        {
+            this.Data = new AvailabilityData();
+            this.context = new TelemetryContext(this.Data.properties);            
+            this.Data.testTimeStamp = testTimeStamp.ToString("o", CultureInfo.InvariantCulture);
```

Comment:
>testTimeStamp [](start = 22, length = 13)

I do not think we need a separate timestamp. That would be confusing and inconsistent with other types. Do you have scenarios in UI where you need both timestamps and they are expected to be different? I do not expect customers to guess that we have a separate field just for this type when they create queries in AI.